plugins/cpp_reparse/service/src/cppreparseservice.cpp

```diff
@@ -92,6 +92,58 @@ void CppReparseServiceHandler::getAsHTML(
   return_ = htmlFactory.str();
 }
 
+void CppReparseServiceHandler::getAsHTMLForNode(
+  std::string& return_,
+  const core::AstNodeId& nodeId_)
+{
+  if (!isEnabled())
+  {
+    return_ = "Reparse capabilities has been disabled at server start.";
+    return;
+  }
+
+  model::CppAstNodePtr astNode;
+  _transaction([&, this](){
+    astNode = _db->query_one<model::CppAstNode>(
+      AstQuery::id == std::stoull(nodeId_));
+    astNode->location.file.load();
+  });
+
+  if (!astNode)
+  {
+    LOG(error) << "No location file found for AST node #" << nodeId_;
+    return_ = "Invalid AST node given, no related file could be found.";
```

Comment:
We have a `common.InvalidId` exception in `common.thrift` for signaling this error.

Reply:
Handling errors, headers and such is added as a `TODO` in the Thrift file. It will be done in a following patch some day.